WalletWasabi/Affiliation/RoundData.cs

```diff
@@ -0,0 +1,44 @@
+using NBitcoin;
+using System.Collections.Generic;
+using System.Linq;
+using WalletWasabi.WabiSabi.Backend.Rounds;
+
+namespace WalletWasabi.Affiliation;
+
+public class RoundData
+{
+	public RoundData(RoundParameters roundParameters)
+	{
+		RoundParameters = roundParameters;
+	}
+
+	private RoundParameters RoundParameters { get; }
+	private Dictionary<OutPoint, AffiliateInput> AffiliateInputsByOutpoint = new();
+
+	public void AddInputCoin(Coin coin, bool isCoordinationFeeExempted)
+	{
+		AffiliateInputsByOutpoint[coin.Outpoint] = 
+			new AffiliateInput(
+				coin.Outpoint,
+				coin.ScriptPubKey,
+				AffiliationConstants.DefaultAffiliationFlag,
+				isCoordinationFeeExempted || IsNoFee(coin.Amount));
+	}
+
+	public void AddInputAffiliationFlag(Coin coin, string affiliationFlag)
+	{
+		var inputData = AffiliateInputsByOutpoint[coin.Outpoint];
+		AffiliateInputsByOutpoint[coin.Outpoint] = inputData with { AffiliationFlag = affiliationFlag };
```

Comment:
It's too late. `afiliationFlag` is now part of a public contract and cannot be changed everywhere. I mean, we could rename internally but that can lead to have more than one name of the same concept.

Reply:
If by public contract you mean that it's used by Trezor Suite, then yeah, it's some complexity, but we can deal with the name change as long as we coordinate on the deployment. If you do decide to change it, then we should use the opportunity to review all the other names.

Reply:
ok, lets do that.

Reply:
In that case, as I already stated, please take a moment to review all the other names in this PR, so that we do it all in one go. Yesterday Lucas and I were discussing that `coinjoin_request` might be too trezor-specific and could be changed to something more generic like `affiliate_data`.

Next Wednesday 22nd February we are organizing a final community test starting at 18:00 CET. If this PR is merged by then, then it would be great to use this opportunity to stress-test the final solution. However, if there are breaking changes like these, then I'd really like to do an internal integration test 48 hours before the community test, i.e. have all the changes in by Monday evening.

Reply:
Regarding the breaking changes we will need even more time. 
Best to have it today.. but latest till the end of a week. We would like to build new version for community testing on Monday morning. 
**Is it possible?** 🙏